### ISSUE TITLE
Single-Wire CAN issues

### ISSUE THREAD

**Issue by pd0wm** [commaai/panda#1551]
I noticed a few issues when trying to communicate with an external device that uses SW-CAN using pin 1 of a white panda.

- Baud rate is 2x too high
- Setting the speed to a different baudrate has no effect, it's stuck on 2x 33.3kbit/s
- The panda does not seem to Ack or receive any messages

Scope screenshot where panda is sending messages. Note baud rate is 2x too high, and ACK from external device. Panda will correctly retry sending the message if an ACK is not received. Yellow = RX, Purple TX, measured behind the `NVC7356` SW-CAN transceiver on the external device.
![ScreenImg (2)](https://github.com/commaai/panda/assets/1314752/a1b53107-5b68-40bb-a0b1-f7cdc7f9167a)

Scope screenshot where external device is sending message (in single shot mode to prevent endless re-transmission). No ack, and no messages received from the Python side. Tried sending at both 33.3kbit/s and 66.6kbit/s in case reception was using the correct baud rate.
![ScreenImg (7)](https://github.com/commaai/panda/assets/1314752/db751d45-b578-4459-8c8d-aedefcae34c7)

Code used to send messages
```python
p = Panda()
p.set_safety_mode(Panda.SAFETY_ALLOUTPUT)
p.can_send(0x0, b"\xaa\xaa", 3)
```

Am I doing something wrong to receive messages? I'm following https://github.com/commaai/panda/blob/master/tests/gmlan_harness_test.py, but I think that uses bit-banged CAN on a regular CAN bus instead of single wire CAN.

**Comment by adeebshihadeh** [comment]
Sounds like ti's using the bitbanged GMLAN for some reason: https://github.com/commaai/panda/blob/6e1d35b03331529829e2075d3491f2693034b6e9/board/drivers/gmlan_alt.h#L133.

**Comment by briskspirit** [comment]
@pd0wm seems like we can close the issue?

**Comment by adeebshihadeh** [comment]
How was this fixed?

**Comment by pd0wm** [comment]
Yes, clocks are fixed now. The other issues I mentioned just aren’t supported by but banged gmlan.

I haven’t been able to get that multiplexing working yet, but will make a new issue if necessary.
